### PR TITLE
Deprecated usage of binary fields whose length exceeds the platform maximum

### LINKED ISSUE #3187: Deprecate usage of binary fields whose length exceeds maximum
As of #3179 (`3.0.0`), `BLOB`s are no longer used to represent binary fields whose length exceeds maximum on a given platform. This logic should be documented as deprecated in the next minor release. If technically possible, deprecation notices should be triggered.

### PR DESCRIPTION
**Pull request by morozov**
Closes #3187.

### CODE REVIEW

File: lib/Doctrine/DBAL/Platforms/AbstractPlatform.php
Comment (Majkl578) [inline lib/Doctrine/DBAL/Platforms/AbstractPlatform.php]: Can we also add a hint here? _"use getBlobTypeDeclarationSQL() instead"_

**Review by Majkl578 — DISMISSED** [review]

File: lib/Doctrine/DBAL/Platforms/AbstractPlatform.php
Comment (morozov) [reply lib/Doctrine/DBAL/Platforms/AbstractPlatform.php]: It may be confusing. Besides using a blob, a user may want to reduce the length of the field. Also, most likely, a user is not calling this method directly, it's called based on their entity definitions.

What about: reduce the field size or use a BLOB field instead?

Comment (Majkl578) [reply lib/Doctrine/DBAL/Platforms/AbstractPlatform.php]: Sounds good. 👍

**Review by Ocramius — DISMISSED** [review]
LGTM 👍

File: lib/Doctrine/DBAL/Platforms/AbstractPlatform.php
Comment (Majkl578) [reply lib/Doctrine/DBAL/Platforms/AbstractPlatform.php]: Hah @Ocramius was too fast. 😂

**Comment by morozov** [comment]
@Majkl578 @Ocramius did you guys just see a race condition here? I force-pushed the updated branch a moment before the pull request got merged, it invalidated the review but still let the merged. Both the merge and invalidation happened on the same second: 3:48 PDT.

**Comment by Ocramius** [comment]
Yeah, saw the dismissed review.

Is the current `master` state diffing anything with your local dev branch?

**Comment by Majkl578** [comment]
Yup, noticed, original commit got merged, the new one interestingly only fired ContinuousPHP build which immediately failed.
Nice!


@morozov Can you send the reworded patch in new PR? 👍

**Comment by morozov** [comment]
> Is the current `master` state diffing anything with your local dev branch?

Actually yes (#3192). What I thought, is the changes I force-pushed got merged even if they weren't approved. So, false alarm.

**Pull request merged**
